### PR TITLE
BUG: Enforce correct encoding in stata

### LINKED ISSUE #15723: ERR: validate encoding on to_stata
It seems ``pandas`` in ``python3.5`` causes issues due to encoding. For example the following generates a corrupt output file

```python
import pandas as pd
df1 = pd.DataFrame(np.array([1,2,3,4]), columns=['var1'])
df1.to_stata('corrupt.dta', write_index=False, encoding='utf8')
```

while 

```python
df1.to_stata('not-corrupt.dta', write_index=False)
```

generates a correct file. I imagine this may be due to use of encoding and the difference in the treatment between ``python 2`` and ``python 3``, which breaks compatibility of scripts across ``python`` versions. I guess it would be nice if it does not take this option into account on ``python 3``, unless the error is caused by something else.

### PR DESCRIPTION
**Pull request by bashtage**
Ensure StataReader and StataWriter have the correct encoding.
Standardized default encoding to 'latin-1'

closes #15723

 - [x] closes #15723
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
@bashtage side issue. We finally have 32-bit daily wheels.

https://travis-ci.org/MacPython/pandas-wheels/jobs/213101390

so this one came up in testing (I am working fixing the rest). Unfort no easy way to test this as has to be on master. But if you have a fix can merge.

```
/venv/local/lib/python2.7/site-packages/pandas/tests/io/test_stata.py:247: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/venv/local/lib/python2.7/site-packages/pandas/tests/io/test_stata.py:88: in read_dta
    return read_stata(file, convert_dates=True)
/venv/local/lib/python2.7/site-packages/pandas/io/stata.py:173: in read_stata
    data = reader.read()
/venv/local/lib/python2.7/site-packages/pandas/io/stata.py:1526: in read
    data = self._insert_strls(data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <pandas.io.stata.StataReader object at 0xede213ec>
data =       x    y            z
0   1.0  abc   4294967299
1   3.0  cba   8589934595
2  93.0       12884901891
    def _insert_strls(self, data):
        if not hasattr(self, 'GSO') or len(self.GSO) == 0:
            return data
        for i, typ in enumerate(self.typlist):
            if typ != 'Q':
                continue
>           data.iloc[:, i] = [self.GSO[k] for k in data.iloc[:, i]]
E           KeyError: 4294967299
/venv/local/lib/python2.7/site-packages/pandas/io/stata.py:1626: KeyError
```

File: pandas/io/stata.py [hunk 1]
```diff
@@ -816,9 +819,14 @@ def get_base_missing_value(cls, dtype):
 
 
 class StataParser(object):
-    _default_encoding = 'iso-8859-1'
+    _default_encoding = 'latin-1'
+
+    def __init__(self, encoding='latin-1'):
+
+        if encoding not in VALID_ENCODINGS:
+            raise ValueError('Unknown encoding. Only latin-1 and  ascii '
```
Comment (jreback) [inline pandas/io/stata.py]: extra space before ascii

File: pandas/tests/io/test_stata.py [hunk 2]
```diff
@@ -1276,3 +1276,9 @@ def test_out_of_range_float(self):
                 original.to_stata(path)
             tm.assertTrue('ColumnTooBig' in cm.exception)
             tm.assertTrue('infinity' in cm.exception)
+
+    def test_invalid_encoding(self):
+        original = self.read_csv(self.csv3)
```
Comment (jreback) [inline pandas/tests/io/test_stata.py]: can you add the commit number as a comment

**Review by jreback — APPROVED** [review]
trivial comments. after its green you can push (and just ping me to merge)

**Comment by bashtage** [comment]
@jreback I rolled a fix for the 32bit issue into this PR.  I can separate if you prefer.

**Comment by jreback** [comment]
@bashtage totally fine here. thanks! ping on green.

**Comment by jreback** [comment]
thanks! always a pleasure @bashtage

**Comment by jreback** [comment]
@bashtage looks like this fixed the 32-bit error as well! https://travis-ci.org/MacPython/pandas-wheels/jobs/213622884

thanks (of course other errors which I am working on ......)

**Pull request closed without merging**